scripts/workers/evaluator.js

```diff
@@ -0,0 +1,36 @@
+"use strict";
+
+// Initialize some global variable to use directly inside eval()
+const abs = Math.abs;
+const floor = Math.floor;
+const ceil = Math.ceil;
+
+const sin = Math.sin;
+const cos = Math.cos;
+const tan = Math.tan;
+
+const log = Math.log;
+
+// Handle message events from main script.
+globalThis.addEventListener("message", function(event) {
+    try {
+        let result = eval(event.data);
+
+        // Handle divide by 0
+        if (Math.abs(result) === Infinity) {
+            throw new Error("This operation is not allowed");
+        }
+
+        globalThis.postMessage({
+            success: true,
+            q: event.data,
+            result
+        });
+    } catch (error) {
+        globalThis.postMessage({
+            success: false,
+            q: event.data,
+            error
```

Comment:
Can we convert error to string to ensure that it always be available in string format.

```suggestion
            error: error.toString()
```

Reply:
Fixed it.
https://github.com/dhruvdhaduk-simform/scientific-calc/pull/3/commits/d9fe68a6dd9b2987e47c6c845269c90fd685995f